src/main/scala/reporting/DomainInterpreter.scala

```diff
@@ -0,0 +1,51 @@
+package viper.silicon.reporting
+import viper.silicon.state.terms._
+import viper.silicon.state.Identifier
+
+/**
+  * the most Generic form of interpreter
+  *  @param	E : input type
+  *  @param I : info object type
+  *  @param	T : result type
+  */
+trait AbstractInterpreter[E, I, T] { 
+	def interpret(entry: E, info: I): T
```

Comment:
Please stick with an indentation of 2 (instead of 4). Probably affects several files.